src/mds/CInode.cc

```diff
@@ -5212,6 +5330,46 @@ mds_rank_t CInode::get_export_pin(bool inherit) const
   return MDS_RANK_NONE;
 }
 
+double CInode::get_export_ephemeral_random_pin(bool inherit) const
+{
+  /* An inode that is export pinned may not necessarily be a subtree root, we
+   * need to traverse the parents. A base or system inode cannot be pinned.
+   * N.B. inodes not yet linked into a dir (i.e. anonymous inodes) will not
+   * have a parent yet.
+   */
+  const CInode *in = this;
+  while (true) {
+    if (in->is_system())
+      break;
+    const CDentry *pdn = in->get_parent_dn();
+    if (!pdn)
+      break;
+    // ignore export pin for unlinked directory
+    if (in->get_inode().nlink == 0)
+      break;
+    // Check if the inode is projected/dirty
+    if (in->is_projected() && in->get_projected_inode()->export_ephemeral_random_pin >= 0)
+      return in->get_projected_inode()->export_ephemeral_random_pin;
```

Comment:
Don't look at the projected value. Use the same logic as export pins. See fix here: 93cbee3dd9c1446a027e7673645a1856ea68637a